asana/asana.go

```diff
@@ -28,8 +28,18 @@ var defaultOptFields = map[string][]string{
 }
 
 type (
+	//Doer interface used for doing http calls.
+	//Use it as point of setting Auth header or custom status code error handling.
+	Doer interface {
+		Do(req *http.Request) (*http.Response, error)
+	}
```

Comment:
Thanks for the explanation.

This sounds reasonable, but I also slightly worry this might not be the most optimal solution for the problem you're trying to solve. Having `Doer` be an interface lets you pass in a middleware-augmented `*http.Client` of sorts, but you still have to create a new `asana.NewClient` for each one...

It's also interesting that go-github, a very popular library, manages to get by without `Doer` interface. /cc @willnoris, do you know how one would do that with go-github?

I'm not opposed to this going in @anpryl, I just wanted to say that it seems a little more questionable to me, and perhaps we can find a better way to effectively be able to augment certain `*http.Request`s. But maybe not, and this is fine.